Ptt/Flows/LoginFlow/Controllers/LoginSubView.swift

```diff
@@ -191,33 +134,29 @@ extension LoginViewController {
         var account = ""
         var passwd = ""
 
-        if let tf = self.tfUsername.view as? UITextField, let tfText = tf.text {
+        if let tfText = self.tfUsername.text {
             account = tfText
         }
-        if let tf = self.tfPassword.view as? UITextField, let tfText = tf.text {
+        if let tfText = self.tfPassword.text {
             passwd = tfText
         }
 
         // block by btnLogin.isEnabled
         // todo: fix
-        if  account.isEmpty {
-            if let tf = self.tfUsername.view as? LoginTextField {
-                tf.warning(msg: L10n.notFinish)
-            }
+        if account.isEmpty {
+            self.tfUsername.warning(msg: L10n.notFinish)
             return
         }
 
         if passwd.isEmpty {
-            if let tf = self.tfPassword.view as? LoginTextField {
-                tf.warning(msg: L10n.notFinish)
-            }
+            self.tfPassword.warning(msg: L10n.notFinish)
             return
         }
 
         self.btnLogin.isEnabled = false
         APIClient.shared.login(account: account, password: passwd) { result in
             self.btnLogin.isEnabled = true
```

Comment:
`-[UIButton setEnabled:] must be used from main thread only`

Reply:
Thanks! I've fixed the issue.